pkg/api/customization/clustertemplate/cluster_template.go

```diff
@@ -70,10 +87,101 @@ func (w Wrapper) LinkHandler(apiContext *types.APIContext, next types.RequestHan
 }
 
 func (w Wrapper) ClusterTemplateRevisionsActionHandler(actionName string, action *types.Action, apiContext *types.APIContext) error {
-	if actionName != "listquestions" {
-		return httperror.NewAPIError(httperror.NotFound, "not found")
+
+	canUpdateClusterTemplateRevision := func() bool {
+		revision := map[string]interface{}{
+			"id": apiContext.ID,
+		}
+
+		return apiContext.AccessControl.CanDo(v3.ClusterTemplateRevisionGroupVersionKind.Group, v3.ClusterTemplateRevisionResource.Name, "update", apiContext, revision, apiContext.Schema) == nil
+	}
+
+	switch actionName {
+	case "disable":
+		if !canUpdateClusterTemplateRevision() {
+			return httperror.NewAPIError(httperror.PermissionDenied, "can not access clusterTemplateRevision")
+		}
+		return w.disableRevision(actionName, action, apiContext)
+	case "enable":
+		if !canUpdateClusterTemplateRevision() {
+			return httperror.NewAPIError(httperror.PermissionDenied, "can not access clusterTemplateRevision")
+		}
+		return w.enableRevision(actionName, action, apiContext)
+	case "listquestions":
+		return w.listRevisionQuestions(actionName, action, apiContext)
+
 	}
+	return httperror.NewAPIError(httperror.NotFound, "not found")
+}
+
+func (w Wrapper) disableRevision(actionName string, action *types.Action, apiContext *types.APIContext) error {
+
+	revision, err := w.loadRevision(apiContext)
+	if err != nil {
+		return httperror.WrapAPIError(err, httperror.ServerError, "failed to load clusterTemplateRevision")
+	}
+
+	if !*revision.Spec.Enabled {
+		apiContext.WriteResponse(http.StatusNoContent, map[string]interface{}{})
+		return nil
+	}
+
+	revisionCopy := revision.DeepCopy()
+	enabled := false
+	revisionCopy.Spec.Enabled = &enabled
+
+	_, err = w.ClusterTemplateRevisions.Update(revisionCopy)
+	if err != nil {
```

Comment:
Do we want to check for conflict error here and retry?

Reply:
I don't think we handle conflict errors in usual update requests, but just return error on the client request indicating the operation failed.

Reply:
will be handling the conflict by looking up from the store